web-app/src/app/(landing)/(auth)/signin/components/form.tsx

```diff
@@ -36,78 +29,41 @@ export default function SignInForm() {
+    const result = await signin(values);
 
-    const result = await signin(formData);
-
-    if (result.error) {
+    if (result.success) {
+      toast.success(t("success"));
+      router.push("/dashboard");
+    } else {
       toast.error(t("error"));
-      return;
     }
-
-    toast.success(t("success"));
-    router.push("/dashboard");
   };
 
   return (
-    <Form {...form}>
-      <form onSubmit={form.handleSubmit(onSubmit)}>
-        <fieldset
-          disabled={form.formState.isSubmitting}
-          className="flex flex-col gap-6"
-        >
-          {signInFormData.map(
-            ({ name, labelKey, placeholderKey, type, descriptionKey }) => (
-              <FormField
-                key={name}
-                control={form.control}
-                name={name}
-                render={({ field }) => (
-                  <FormItem>
-                    <FormLabel>{labelKey && t(labelKey)}</FormLabel>
-                    <FormControl>
-                      <Input
-                        placeholder={placeholderKey && t(placeholderKey)}
-                        type={type || "text"}
-                        {...field}
-                      />
-                    </FormControl>
-                    {descriptionKey && (
-                      <FormDescription>{t(descriptionKey)}</FormDescription>
-                    )}
-                    <FormMessage />
-                  </FormItem>
-                )}
-              />
-            ),
-          )}
-          <div className="flex flex-col items-center justify-between gap-2 sm:flex-row">
-            <Button
-              type="submit"
-              className="w-full sm:w-auto"
-              disabled={form.formState.isSubmitting}
-            >
-              {form.formState.isSubmitting && (
-                <Loader2 className="mr-2 h-4 w-4 animate-spin" />
-              )}
-              {t("submit")}
-            </Button>
-            <div className="text-sm">
-              <span className="text-muted-foreground">
-                {t("ForgotPassword.text")}{" "}
-              </span>
-              <Link
-                href="/forgot-password"
-                className="text-primary font-medium hover:underline"
-              >
-                {t("ForgotPassword.link")}
-              </Link>
-            </div>
-          </div>
-        </fieldset>
-      </form>
-    </Form>
+    <AuthForm
+      form={form}
+      formData={signInFormData}
+      namespace="Auth.Pages.SignIn.Form"
+      onSubmit={onSubmit}
+    >
+      <div className="flex flex-col items-center justify-between gap-2 sm:flex-row">
+        <SubmitButton
+          form={form}
+          label={t("submit")}
+          className="w-full sm:w-auto"
+        />
+        <div className="text-sm">
+          <span className="text-muted-foreground">
+            {t("ForgotPassword.text")}{" "}
+          </span>
+          <Link
+            href="/forgot-password"
```

Comment:
do we want to also preserve the email here if entered before?

optional feature, and maybe not for now, but maybe we can collect such ideas in a list for later, what do you think?

Reply:
Nice idea... e99eac68f7829818958351657e6f793808723f63